### PR TITLE
Fix issue #2624

### PR DESCRIPTION
**Pull request by JTPetter**
fixes https://github.com/jasp-stats/jasp-test-release/issues/2624

### CODE REVIEW

**Review by maltelueken — APPROVED** [review]
Looks good!

Three tests are failing on Mac, is that expected?

**Comment by JTPetter** [comment]
But looks like I need another review to merge...

**Comment by koenderks** [comment]
It is probably best to add testthat::skip_on_os("mac") to the failing test and then make another test specifically for mac with the NaN while adding testthat::skip_on_os(c("windows", "linux")). Preferably we have all successful tests even if there is a NaN. So then you have one test for windows and linux and one for mac.

File: R/doeFactorial.R [hunk 1]
```diff
@@ -337,7 +337,7 @@ doeFactorial <- function(jaspResults, dataset, options, ...) {
   df <- .doeRsmCategorical2df(options[["categoricalVariables"]])
   designSpec <- .doeFactorialGetSelectedDesign(jaspResults, options)
   if (length(designSpec) == 0) {
-    return()
+    stop("No generator provided.")
```
Comment (koenderks) [inline R/doeFactorial.R:340]: jaspBase:::.quitAnalysis()?

**Review by koenderks — APPROVED** [review]

**Comment by JTPetter** [comment]
I did not know about he skipping. Thank you! I added it.

**Pull request merged**